src/Hive.Versioning/Version.cs

```diff
@@ -516,6 +524,23 @@ private static bool TryParse(ref StringPart text, bool checkLength, [MaybeNullWh
             return TryParse(ref errors, ref text, checkLength, out version); // don't do error reporting
         }
 
+        private static bool TryParse(ref StringPart text, bool checkLength, [MaybeNullWhen(false)] out Version version, [MaybeNullWhen(true)] out string error)
+        {
+            var errors = new ErrorState(text);
```

Comment:
Is there value to having this in a `using` block instead? That'd manage the `Dispose` for you instead of having to be explicit, right?

Reply:
IIRC there were some issues with that back when I wrote this, possibly because the language version being used doesn't support pattern-based `IDisposable`. We could upgrade the language version, but I feel like that's a separate PR.